## Practice Exercise 2

In this assignment, you will try to find some interesting insights into a few movies released between 1916 and 2016, using Python. You will have to download a movie dataset, write Python code to explore the data, gain insights into the movies, actors, directors, and collections, and submit the code.

#### Some tips before starting the assignment

1. Identify the task to be performed correctly, and only then proceed to write the required code. Don’t perform any incorrect analysis or look for information that isn’t required for the assignment.
2. In some cases, the variable names have already been assigned, and you just need to write code against them. In other cases, the names to be given are mentioned in the instructions. We strongly advise you to use the mentioned names only.
3. Always keep inspecting your data frame after you have performed a particular set of operations.
4. There are some checkpoints given in the IPython notebook provided. They're just useful pieces of information you can use to check if the result you have obtained after performing a particular task is correct or not.
5. Note that you will be asked to refer to documentation for solving some of the questions. That is done on purpose for you to learn new commands and also how to use the documentation.

In [1]:
# Import the numpy and pandas packages

#import numpy as np
import pandas as pd

### Task 1: Reading and Inspection

**Subtask 1.1: Import and read**

Import and read the movie database. Store it in a variable called `movies`.

In [2]:
# Write your code for importing the csv file here
movies = pd.read_csv("Data/Movies.csv")
movies

color      director_name  num_critic_for_reviews  duration  \
0     Color      James Cameron                   723.0     178.0   
1     Color     Gore Verbinski                   302.0     169.0   
2     Color         Sam Mendes                   602.0     148.0   
3     Color  Christopher Nolan                   813.0     164.0   
4     Color     Andrew Stanton                   462.0     132.0   
...     ...                ...                     ...       ...   
3848  Color      Shane Carruth                   143.0      77.0   
3849  Color   Neill Dela Llana                    35.0      80.0   
3850  Color   Robert Rodriguez                    56.0      81.0   
3851  Color       Edward Burns                    14.0      95.0   
3852  Color           Jon Gunn                    43.0      90.0   

      director_facebook_likes  actor_3_facebook_likes        actor_2_name  \
0                         0.0                   855.0    Joel David Moore   
1                       563.0                  1000.0       Orlando Bloom   
2                         0.0                   161.0        Rory Kinnear   
3                     22000.0                 23000.0      Christian Bale   
4                       475.0                   530.0     Samantha Morton   
...                       ...                     ...                 ...   
3848                    291.0                     8.0      David Sullivan   
3849                      0.0                     0.0     Edgar Tancangco   
3850                      0.0                     6.0     Peter Marquardt   
3851                      0.0                   133.0  Caitlin FitzGerald   
3852                     16.0                    16.0    Brian Herzlinger   

      actor_1_facebook_likes        gross  \
0                     1000.0  760505847.0   
1                    40000.0  309404152.0   
2                    11000.0  200074175.0   
3                    27000.0  448130642.0   
4                      640.0   73058679.0   
...                      ...          ...   
3848                   291.0     424760.0   
3849                     0.0      70071.0   
3850                   121.0    2040920.0   
3851                   296.0       4584.0   
3852                    86.0      85222.0   

                                   genres  ... num_user_for_reviews language  \
0         Action|Adventure|Fantasy|Sci-Fi  ...               3054.0  English   
1                Action|Adventure|Fantasy  ...               1238.0  English   
2               Action|Adventure|Thriller  ...                994.0  English   
3                         Action|Thriller  ...               2701.0  English   
4                 Action|Adventure|Sci-Fi  ...                738.0  English   
...                                   ...  ...                  ...      ...   
3848                Drama|Sci-Fi|Thriller  ...                371.0  English   
3849                             Thriller  ...                 35.0  English   
3850  Action|Crime|Drama|Romance|Thriller  ...                130.0  Spanish   
3851                         Comedy|Drama  ...                 14.0  English   
3852                          Documentary  ...                 84.0  English   

          country  content_rating       budget  title_year  \
0             USA           PG-13  237000000.0      2009.0   
1             USA           PG-13  300000000.0      2007.0   
2              UK           PG-13  245000000.0      2015.0   
3             USA           PG-13  250000000.0      2012.0   
4             USA           PG-13  263700000.0      2012.0   
...           ...             ...          ...         ...   
3848          USA           PG-13       7000.0      2004.0   
3849  Philippines       Not Rated       7000.0      2005.0   
3850          USA               R       7000.0      1992.0   
3851          USA       Not Rated       9000.0      2011.0   
3852          USA              PG       1100.0      2004.0   

     actor_2_face

**Subtask 1.2: Inspect the dataframe**

Inspect the dataframe's columns, shapes, variable types etc.

In [3]:
# Write your code for inspection here
print(movies.columns, movies.shape, "\n", movies.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3853 entries, 0 to 3852
Data columns (total 28 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   color                      3851 non-null   object 
 1   director_name              3853 non-null   object 
 2   num_critic_for_reviews     3852 non-null   float64
 3   duration                   3852 non-null   float64
 4   director_facebook_likes    3853 non-null   float64
 5   actor_3_facebook_likes     3847 non-null   float64
 6   actor_2_name               3852 non-null   object 
 7   actor_1_facebook_likes     3853 non-null   float64
 8   gross                      3853 non-null   float64
 9   genres                     3853 non-null   object 
 10  actor_1_name               3853 non-null   object 
 11  movie_title                3853 non-null   object 
 12  num_voted_users            3853 non-null   int64  
 13  cast_total_facebook_likes  3853 non-null   int64

#### <font color='red'>Question 1: How many rows and columns are present in the dataframe? </font>
-  <font color='red'>(3821, 26)</font>
-  <font color='red'>(3879, 28)</font>
-  <font color='red'>(3853, 28)</font>
-  <font color='red'>(3866, 26)</font>

#### <font color='red'>Question 2: How many columns have null values present in them? Try writing a code for this instead of counting them manually.</font>

-  <font color='red'>3</font>
-  <font color='red'>6</font>
-  <font color='red'>9</font>
-  <font color='red'>12</font>

In [8]:
len(movies.columns[movies.isna().any()].tolist())

12

### Task 2: Cleaning the Data

**Subtask 2.1: Drop unecessary columns**

For this assignment, you will mostly be analyzing the movies with respect to the ratings, gross collection, popularity of movies, etc. So many of the columns in this dataframe are not required. So it is advised to drop the following columns.
-  color
-  director_facebook_likes
-  actor_1_facebook_likes
-  actor_2_facebook_likes
-  actor_3_facebook_likes
-  actor_2_name
-  cast_total_facebook_likes
-  actor_3_name
-  duration
-  facenumber_in_poster
-  content_rating
-  country
-  movie_imdb_link
-  aspect_ratio
-  plot_keywords

In [9]:
# Check the 'drop' function in the Pandas library - dataframe.drop(list_of_unnecessary_columns, axis = )
# Write your code for dropping the columns here. It is advised to keep inspecting the dataframe after each set of operations

movies = movies.drop(['color', 'director_facebook_likes', 'actor_1_facebook_likes', 'actor_2_facebook_likes', 'actor_3_facebook_likes',
                        'actor_2_name', 'cast_total_facebook_likes', 'actor_3_name', 'duration', 'facenumber_in_poster', 'content_rating', 
                        'country', 'movie_imdb_link', 'aspect_ratio', 'plot_keywords' ],axis=1)
len(movies.columns.tolist())

13

#### <font color='red'>Question 3: What is the count of columns in the new dataframe? </font>
-  <font color='red'>10</font>
-  <font color='red'>13</font>
-  <font color='red'>15</font>
-  <font color='red'>17</font>

**Subtask 2.2: Inspect Null values**

As you have seen above, there are null values in multiple columns of the dataframe 'movies'. Find out the percentage of null values in each column of the dataframe 'movies'. 

In [16]:
# Write you code here
movies.isna().sum()

director_name             0
num_critic_for_reviews    1
gross                     0
genres                    0
actor_1_name              0
movie_title               0
num_voted_users           0
num_user_for_reviews      0
language                  3
budget                    0
title_year                0
imdb_score                0
movie_facebook_likes      0
dtype: int64

#### <font color='red'>Question 4: Which column has the highest percentage of null values? </font>
-  <font color='red'>language</font>
-  <font color='red'>genres</font>
-  <font color='red'>num_critic_for_reviews</font>
-  <font color='red'>imdb_score</font>

**Subtask 2.3: Fill NaN values**

You might notice that the `language` column has some NaN values. Here, on inspection, you will see that it is safe to replace all the missing values with `'English'`.

In [19]:
# Write your code for filling the NaN values in the 'language' column here
import numpy as np
print("Before:", movies[movies['language'] == 'English'].shape)
movies['language'] = movies['language'].replace(np.nan,'English')
print("After:", movies[movies['language'] == 'English'].shape)

Before: (3671, 13)
After: (3674, 13)


#### <font color='red'>Question 5: What is the count of movies made in English language after replacing the NaN values with English? </font>
-  <font color='red'>3670</font>
-  <font color='red'>3674</font>
-  <font color='red'>3668</font>
-  <font color='red'>3672</font>

### Task 3: Data Analysis

**Subtask 3.1: Change the unit of columns**

Convert the unit of the `budget` and `gross` columns from `$` to `million $`.

In [23]:
# Write your code for unit conversion here
movies[['budget','gross']] = movies[['budget','gross']] / 1000000
movies[['budget','gross']] 

budget         gross
0     2.370000e-04  7.605058e-04
1     3.000000e-04  3.094042e-04
2     2.450000e-04  2.000742e-04
3     2.500000e-04  4.481306e-04
4     2.637000e-04  7.305868e-05
...            ...           ...
3848  7.000000e-09  4.247600e-07
3849  7.000000e-09  7.007100e-08
3850  7.000000e-09  2.040920e-06
3851  9.000000e-09  4.584000e-09
3852  1.100000e-09  8.522200e-08

[3853 rows x 2 columns]

**Subtask 3.2: Find the movies with highest profit**

   1. Create a new column called `profit` which contains the difference of the two columns: `gross` and `budget`.
   2. Sort the dataframe using the `profit` column as reference. (Find which command can be used here to sort entries from the documentation)
   3. Extract the top ten profiting movies in descending order and store them in a new dataframe - `top10`

In [26]:
# Write your code for creating the profit column here
movies['profit'] = movies['gross'] - movies['budget']
movies = movies.sort_values(by=['profit'], ascending=False)
movies.head()

director_name  num_critic_for_reviews     gross  \
0        James Cameron                   723.0  0.000761   
28     Colin Trevorrow                   644.0  0.000652   
25       James Cameron                   315.0  0.000659   
2704      George Lucas                   282.0  0.000461   
2748  Steven Spielberg                   215.0  0.000435   

                                genres         actor_1_name  \
0      Action|Adventure|Fantasy|Sci-Fi          CCH Pounder   
28    Action|Adventure|Sci-Fi|Thriller  Bryce Dallas Howard   
25                       Drama|Romance    Leonardo DiCaprio   
2704   Action|Adventure|Fantasy|Sci-Fi        Harrison Ford   
2748                     Family|Sci-Fi         Henry Thomas   

                              movie_title  num_voted_users  \
0                                 Avatar            886204   
28                        Jurassic World            418214   
25                               Titanic            793059   
2704  Star Wars: Episode IV - A New Hope            911097   
2748          E.T. the Extra-Terrestrial            281842   

      num_user_for_reviews language    budget  title_year  imdb_score  \
0                   3054.0  English  0.000237      2009.0         7.9   
28                  1290.0  English  0.000150      2015.0         7.0   
25                  2528.0  English  0.000200      1997.0         7.7   
2704                1470.0  English  0.000011      1977.0         8.7   
2748                 515.0  English  0.000010      1982.0         7.9   

      movie_facebook_likes    profit  
0                    33000  0.000524  
28                  150000  0.000502  
25                   26000  0.000459  
2704                 33000  0.000450  
2748                 34000  0.000424

In [ ]:
# Write your code for sorting the dataframe here


In [ ]:
top10 = # Write your code to get the top 10 profiting movies here

**Checkpoint:** You might spot two movies directed by `James Cameron` in the list.

#### <font color='red'>Question 6: Which movie is ranked 5th from the top in the list obtained? </font>
-  <font color='red'>E.T. the Extra-Terrestrial</font>
-  <font color='red'>The Avengers</font>
-  <font color='red'>The Dark Knight</font>
-  <font color='red'>Titanic</font>

**Subtask 3.3: Find IMDb Top 250**

Create a new dataframe `IMDb_Top_250` and store the top 250 movies with the highest IMDb Rating (corresponding to the column: `imdb_score`). Also make sure that for all of these movies, the `num_voted_users` is greater than 25,000. 

Also add a `Rank` column containing the values 1 to 250 indicating the ranks of the corresponding films.

In [36]:
# Write your code for extracting the top 250 movies as per the IMDb score here. Make sure that you store it in a new dataframe 
# and name that dataframe as 'IMDb_Top_250'
IMDb_Top_250 = movies.sort_values(by="imdb_score", ascending=False).reset_index()
IMDb_Top_250 = IMDb_Top_250.iloc[:250,:]
IMDb_Top_250['Rank'] = IMDb_Top_250.index + 1
IMDb_Top_250 


index         director_name  num_critic_for_reviews         gross  \
0     1795        Frank Darabont                   199.0  2.834147e-05   
1     3016  Francis Ford Coppola                   208.0  1.348220e-04   
2       64     Christopher Nolan                   645.0  5.333161e-04   
3     2543  Francis Ford Coppola                   149.0  5.730000e-05   
4     3607          Sergio Leone                   181.0  6.100000e-06   
..     ...                   ...                     ...           ...   
245      0         James Cameron                   723.0  7.605058e-04   
246    339         John Lasseter                   191.0  2.458234e-04   
247     67           Jon Favreau                   486.0  3.182982e-04   
248   1502       Nick Cassavetes                   177.0  6.428600e-08   
249     95         Peter Jackson                   645.0  3.030012e-04   

                                        genres       actor_1_name  \
0                                  Crime|Drama     Morgan Freeman   
1                                  Crime|Drama          Al Pacino   
2                  Action|Crime|Drama|Thriller     Christian Bale   
3                                  Crime|Drama     Robert De Niro   
4                                      Western     Clint Eastwood   
..                                         ...                ...   
245            Action|Adventure|Fantasy|Sci-Fi        CCH Pounder   
246  Adventure|Animation|Comedy|Family|Fantasy          Tom Hanks   
247                    Action|Adventure|Sci-Fi  Robert Downey Jr.   
248                              Drama|Romance       Ryan Gosling   
249                          Adventure|Fantasy       Aidan Turner   

                            movie_title  num_voted_users  \
0             The Shawshank Redemption           1689764   
1                        The Godfather           1155770   
2                      The Dark Knight           1676169   
3               The Godfather: Part II            790926   
4       The Good, the Bad and the Ugly            503509   
..                                  ...              ...   
245                             Avatar            886204   
246                        Toy Story 2            385871   
247                           Iron Man            696338   
248                       The Notebook            396396   
249  The Hobbit: An Unexpected Journey            637246   

     num_user_for_reviews language    budget  title_year  imdb_score  \
0                  4144.0  English  0.000025      1994.0         9.3   
1                  2238.0  English  0.000006      1972.0         9.2   
2                  4667.0  English  0.000185      2008.0         9.0   
3                   650.0  English  0.000013      1974.0         9.0   
4                   780.0  Italian  0.000001      1966.0         8.9   
..                    ...      ...       ...         ...         ...   
245                3054.0  English  0.000237      2009.0         7.9   
246                 515.0  English  0.000090      1999.0         7.9   
247                1055.0  English  0.000140      2008.0         7.9   
248                1111.0  English  0.000029      2004.0         7.9   
249                1367.0  English  0.000180      2012.0         7.9   

     movie_facebook_likes    profit  Rank  
0                  108000  0.000003     1  
1                   43000  0.000129     2  
2                   37000  0.000348     3  
3                   14000  0.000044     4  
4                   20000  0.000005     5  
..                    ...       ...   ...  
245                 33000  0.000524   246  
246                     0  0.000156   247  
247                 10000  0.000178   248  
248                 57000 -0.000029   249  
249                166000  0.000123   250  

[250 rows x 16 columns]

In [39]:
bins = [7, 7.5, 8, 8.5, 9, 9.5, 10 ]
s = pd.cut(IMDb_Top_250['imdb_score'], bins=bins).value_counts()
s

(8.0, 8.5]     132
(7.5, 8.0]      90
(8.5, 9.0]      26
(9.0, 9.5]       2
(7.0, 7.5]       0
(9.5, 10.0]      0
Name: imdb_score, dtype: int64

#### <font color='red'>Question 7: Suppose movies are divided into 5 buckets based on the IMDb ratings: </font>
-  <font color='red'>7.5 to 8</font>
-  <font color='red'>8 to 8.5</font>
-  <font color='red'>8.5 to 9</font>
-  <font color='red'>9 to 9.5</font>
-  <font color='red'>9.5 to 10</font>

<font color = 'red'> Which bucket holds the maximum number of movies from *IMDb_Top_250*? </font>

**Subtask 3.4: Find the critic-favorite and audience-favorite actors**

   1. Create three new dataframes namely, `Meryl_Streep`, `Leo_Caprio`, and `Brad_Pitt` which contain the movies in which the actors: 'Meryl Streep', 'Leonardo DiCaprio', and 'Brad Pitt' are the lead actors. Use only the `actor_1_name` column for extraction. Also, make sure that you use the names 'Meryl Streep', 'Leonardo DiCaprio', and 'Brad Pitt' for the said extraction.
   2. Append the rows of all these dataframes and store them in a new dataframe named `Combined`.
   3. Group the combined dataframe using the `actor_1_name` column.
   4. Find the mean of the `num_critic_for_reviews` and `num_user_for_review` and identify the actors which have the highest mean.

In [40]:
# Write your code for creating three new dataframes here
Meryl_Streep = movies[movies['actor_1_name']=='Meryl Streep'] # Include all movies in which Meryl_Streep is the lead

In [48]:
Leo_Caprio = movies[movies['actor_1_name']=='Leonardo DiCaprio'] # Include all movies in which Leo_Caprio is the lead

In [49]:
Leo_Caprio

director_name  num_critic_for_reviews     gross  \
25            James Cameron                   315.0  0.000659   
94        Christopher Nolan                   642.0  0.000293   
859        Steven Spielberg                   194.0  0.000164   
283       Quentin Tarantino                   765.0  0.000163   
173   Alejandro G. Iñárritu                   556.0  0.000184   
431         Martin Scorsese                   490.0  0.000128   
347         Martin Scorsese                   352.0  0.000132   
49             Baz Luhrmann                   490.0  0.000145   
3025           Baz Luhrmann                   490.0  0.000145   
2484           Baz Luhrmann                   106.0  0.000046   
1336        Randall Wallace                    83.0  0.000057   
294         Martin Scorsese                   606.0  0.000117   
1364         Clint Eastwood                   392.0  0.000037   
246         Martin Scorsese                   267.0  0.000103   
1909             Jerry Zaks                    45.0  0.000013   
934             Danny Boyle                   118.0  0.000040   
1046             Sam Mendes                   323.0  0.000023   
1457              Sam Raimi                    63.0  0.000019   
312         Martin Scorsese                   233.0  0.000078   
608            Ridley Scott                   238.0  0.000039   
293            Edward Zwick                   166.0  0.000057   

                                genres       actor_1_name  \
25                       Drama|Romance  Leonardo DiCaprio   
94    Action|Adventure|Sci-Fi|Thriller  Leonardo DiCaprio   
859              Biography|Crime|Drama  Leonardo DiCaprio   
283                      Drama|Western  Leonardo DiCaprio   
173   Adventure|Drama|Thriller|Western  Leonardo DiCaprio   
431                   Mystery|Thriller  Leonardo DiCaprio   
347               Crime|Drama|Thriller  Leonardo DiCaprio   
49                       Drama|Romance  Leonardo DiCaprio   
3025                     Drama|Romance  Leonardo DiCaprio   
2484                     Drama|Romance  Leonardo DiCaprio   
1336                  Action|Adventure  Leonardo DiCaprio   
294       Biography|Comedy|Crime|Drama  Leonardo DiCaprio   
1364             Biography|Crime|Drama  Leonardo DiCaprio   
246                    Biography|Drama  Leonardo DiCaprio   
1909                             Drama  Leonardo DiCaprio   
934           Adventure|Drama|Thriller  Leonardo DiCaprio   
1046                     Drama|Romance  Leonardo DiCaprio   
1457           Action|Thriller|Western  Leonardo DiCaprio   
312                        Crime|Drama  Leonardo DiCaprio   
608              Action|Drama|Thriller  Leonardo DiCaprio   
293           Adventure|Drama|Thriller  Leonardo DiCaprio   

                    movie_title  num_voted_users  num_user_for_reviews  \
25                     Titanic            793059                2528.0   
94                   Inception           1468200                2803.0   
859        Catch Me If You Can            525801                 667.0   
283           Django Unchained            955174                1193.0   
173               The Revenant            406020                1188.0   
431             Shutter Island            786092                 964.0   
347               The Departed            873649                2054.0   
49            The Great Gatsby            362912                 753.0   
3025          The Great Gatsby            362933                 753.0   
2484            Romeo + Juliet            167750                 506.0   
1336  The Man in the Iron Mask            125219                 244.0   
294    The Wolf of Wall Street            780588                1138.0   
1364                  J. Edgar            102728                 279.0   
246                The Aviator            264318                 799.0   
1909             Marvin's Room             20163                  71.0   
934                  The Beach            176169        

In [43]:
Brad_Pitt = movies[movies['actor_1_name']=='Brad Pitt'] # Include all movies in which Brad_Pitt is the lead

In [55]:
# Write your code for combining the three dataframes here
Combined = pd.concat([Meryl_Streep, Leo_Caprio, Brad_Pitt])
Combined

director_name  num_critic_for_reviews         gross  \
1322          David Frankel                   208.0  1.247330e-04   
1471         Sydney Pollack                    66.0  8.710000e-05   
1132            Nora Ephron                   252.0  9.412543e-05   
1514          David Frankel                   234.0  6.353601e-05   
392            Nancy Meyers                   187.0  1.127035e-04   
2500         Phyllida Lloyd                   331.0  2.995944e-05   
1784         Stephen Daldry                   174.0  4.159783e-05   
2793          Robert Altman                   211.0  2.033861e-05   
1038          Curtis Hanson                    42.0  4.681575e-05   
1563          Carl Franklin                    64.0  2.320944e-05   
1390         Robert Redford                   227.0  1.499807e-05   
25            James Cameron                   315.0  6.586723e-04   
94        Christopher Nolan                   642.0  2.925689e-04   
859        Steven Spielberg                   194.0  1.644352e-04   
283       Quentin Tarantino                   765.0  1.628046e-04   
173   Alejandro G. Iñárritu                   556.0  1.836359e-04   
431         Martin Scorsese                   490.0  1.279684e-04   
347         Martin Scorsese                   352.0  1.323734e-04   
49             Baz Luhrmann                   490.0  1.448128e-04   
3025           Baz Luhrmann                   490.0  1.448128e-04   
2484           Baz Luhrmann                   106.0  4.633873e-05   
1336        Randall Wallace                    83.0  5.687637e-05   
294         Martin Scorsese                   606.0  1.168667e-04   
1364         Clint Eastwood                   392.0  3.730495e-05   
246         Martin Scorsese                   267.0  1.026088e-04   
1909             Jerry Zaks                    45.0  1.278251e-05   
934             Danny Boyle                   118.0  3.977860e-05   
1046             Sam Mendes                   323.0  2.287781e-05   
1457              Sam Raimi                    63.0  1.863654e-05   
312         Martin Scorsese                   233.0  7.767964e-05   
608            Ridley Scott                   238.0  3.938044e-05   
293            Edward Zwick                   166.0  5.736626e-05   
383       Steven Soderbergh                   186.0  1.834058e-04   
244              Doug Liman                   233.0  1.863361e-04   
887             Neil Jordan                   120.0  1.052646e-04   
448              David Ayer                   406.0  8.570712e-05   
243       Steven Soderbergh                   198.0  1.255316e-04   
2030  Alejandro G. Iñárritu                   285.0  3.430077e-05   
2428         Andrew Dominik                   414.0  1.493857e-05   
2602             Tony Scott                   122.0  1.228150e-05   
2146    Angelina Jolie Pitt                   131.0  5.310090e-07   
1397        Terrence Malick                   584.0  1.330332e-05   
97            David Fincher                   362.0  1.274908e-04   
646           David Fincher                   315.0  3.702340e-05   
1608         Andrew Dominik                   273.0  3.904982e-06   
579     Jean-Jacques Annaud                    76.0  3.790151e-05   
749         Patrick Gilmore                    98.0  2.628832e-05   
142       Wolfgang Petersen                   220.0  1.332283e-04   
367              Tony Scott                   142.0  2.687100e-08   

                                                 genres       actor_1_name  \
1322                               Comedy|Drama|Romance       Meryl Streep   
1471                            Biography|Drama|Romance       Meryl Streep   
1132                            Biography|Drama|Romance       Meryl Streep   
1514                               Comedy|Drama|Romance       Meryl Streep   
392                                Comedy|Drama|Romance       Meryl Streep   
2500                            Biography|Drama|History       Meryl Streep   
1784             

In [57]:
# Write your code for grouping the combined dataframe here
Combined = Combined.groupby("actor_1_name")['num_critic_for_reviews', 'num_user_for_reviews'].mean()
Combined.sort_values(by='num_user_for_reviews', ascending=False)

ipykernel_launcher:2: FutureWarning: Indexing with multiple keys (implicitly converted to a tuple of keys) will be deprecated, use a list instead.


num_critic_for_reviews  num_user_for_reviews
actor_1_name                                                   
Leonardo DiCaprio              330.190476            914.476190
Brad Pitt                      245.000000            742.352941
Meryl Streep                   181.454545            297.181818

In [58]:
# Write the code for finding the mean of critic reviews and audience reviews here
Combined.sort_values(by='num_critic_for_reviews', ascending=False)

num_critic_for_reviews  num_user_for_reviews
actor_1_name                                                   
Leonardo DiCaprio              330.190476            914.476190
Brad Pitt                      245.000000            742.352941
Meryl Streep                   181.454545            297.181818

#### <font color='red'>Question 8: Which actor is highest rated among the three actors according to the user reviews? </font>
-  <font color='red'>Meryl Streep</font>
-  <font color='red'>Leonardo DiCaprio</font>
-  <font color='red'>Brad Pitt</font>

#### <font color='red'>Question 9: Which actor is highest rated among the three actors according to the critics?</font>
-  <font color='red'>Meryl Streep</font>
-  <font color='red'>Leonardo DiCaprio</font>
-  <font color='red'>Brad Pitt</font>

In [62]:
df = movies

In [ ]:
df.drop(columns=[])